internal/providers/pluginfw/products/permissions/resource_permission.go

```diff
@@ -0,0 +1,652 @@
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("service_principal_name"), &servicePrincipalName)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("permission_level"), &permissionLevel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	w, err := r.Client.WorkspaceClient()
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get workspace client", err.Error())
+		return
+	}
+
+	// Parse the ID to get object ID and principal
+	objectID, _, err := r.parseID(id.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to parse resource ID", err.Error())
+		return
+	}
+
+	// Lock the object to prevent concurrent modifications
+	lockObject(objectID)
+	defer unlockObject(objectID)
+
+	// Get the mapping from the ID
+	mapping, err := permissions.GetResourcePermissionsFromId(objectID)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get resource permissions mapping", err.Error())
+		return
+	}
+
+	// Update the permission using PATCH
+	permLevel := iam.PermissionLevel(permissionLevel.ValueString())
+	idParts := strings.Split(objectID, "/")
+	permID := idParts[len(idParts)-1]
+
+	_, err = w.Permissions.Update(ctx, iam.UpdateObjectPermissions{
+		RequestObjectId:   permID,
+		RequestObjectType: mapping.GetRequestObjectType(),
+		AccessControlList: []iam.AccessControlRequest{
+			{
+				UserName:             userName.ValueString(),
+				GroupName:            groupName.ValueString(),
+				ServicePrincipalName: servicePrincipalName.ValueString(),
+				PermissionLevel:      permLevel,
+			},
+		},
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to update permission", err.Error())
+		return
+	}
+
+	// Read the object identifier field from current state to preserve it
+	var objectFieldValue types.String
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root(mapping.GetField()), &objectFieldValue)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update state using SetAttribute - preserve the object identifier field
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root(mapping.GetField()), objectFieldValue)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("permission_level"), permissionLevel)...)
+}
+
+func (r *PermissionResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Read ID from state using GetAttribute
+	var id types.String
+	resp.Diagnostics.Append(req.State.GetAttribute(ctx, path.Root("id"), &id)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	w, err := r.Client.WorkspaceClient()
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get workspace client", err.Error())
+		return
+	}
+
+	// Parse the ID to get object ID and principal
+	objectID, principal, err := r.parseID(id.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to parse resource ID", err.Error())
+		return
+	}
```

Comment:
Ditto, we shouldn't need to parse the ID to get details about the permission, it should be in state. The only place it may not be in state is in Read (post-import).